decoding/xml_decoder_test.go

```diff
@@ -0,0 +1,305 @@
+package decoding
```

Comment:
Nit `decoding` is probably a little too generic package for this since these are all XML behavior.

I suggest that these XMl files be moved to either the `xml` package in smithy-go similar to `json` and other.  

Though we should consider moving all of the encoding packages until a `encoding` root package for organization.

```
smithy-go
     /encoding
          /json
          /httpbinding
          /xml
```